lib/nexthop.c

```diff
@@ -36,6 +36,63 @@
 DEFINE_MTYPE_STATIC(LIB, NEXTHOP, "Nexthop")
 DEFINE_MTYPE_STATIC(LIB, NH_LABEL, "Nexthop label")
 
+int nexthop_cmp(const struct nexthop *next1, const struct nexthop *next2)
+{
+	int ret;
+	uint32_t n1, n2;
+
+	if (next1->vrf_id < next2->vrf_id)
+		return -1;
+
+	if (next1->vrf_id > next2->vrf_id)
+		return 1;
+
+	if (next1->type < next2->type)
+		return -1;
+
+	if (next1->type > next2->type)
+		return 1;
+
+	switch(next1->type) {
+	case NEXTHOP_TYPE_IPV4:
+		n1 = ntohl(next1->gate.ipv4.s_addr);
+		n2 = ntohl(next2->gate.ipv4.s_addr);
+		if (n1 < n2)
+			return -1;
+		if (n1 > n2)
+			return 1;
+		break;
+	case NEXTHOP_TYPE_IPV6:
+		ret = memcmp(&next1->gate, &next2->gate, sizeof(union g_addr));
+		if (!ret)
```

Comment:
shouldn't it be `(ret != 0)` here (and below) ?

Reply:
Yes, I think it should.